### PR TITLE
feat: RFC-0759: check npm and github for template

### PR DESCRIPTION
**Pull request by blakef**
Summary:
---------

Support RFC-0759 decoupling of template from react-native. This makes use of the new @react-native-community/template project. We're careful to validate against what npm and github show for the state of the react-native project.

Test Plan:
----------

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

File: packages/cli/src/commands/init/init.ts [hunk 1]
```diff
@@ -384,12 +387,62 @@ function checkPackageManagerAvailability(
...
-    options.template === 'react-native-template-typescript';
+    options.template === TEMPLATE_PACKAGE_LEGACY_TYPESCRIPT;
 
   // This allows to correctly retrieve template uri for out of tree platforms.
-  const platform = options.platformName || 'react-native';
+  const platform = options.platformName ?? TEMPLATE_PACKAGE_COMMUNITY;
+
+  let npmVersion: string | undefined;
+
+  if (platform === TEMPLATE_PACKAGE_COMMUNITY) {
```
Comment (cortinico) [inline packages/cli/src/commands/init/init.ts:402]: Q: The Template Versioning still needs to be clarified.
If `version` here is `0.75.0-rc.1`, what are you going to resolve on NPM?

As `https://regsitry.npmjs.org/@react-native-community/template/0.75.0-rc.1` will return 404 most likely

We need the version mapping logic to be included here

Comment (blakef) [reply packages/cli/src/commands/init/init.ts:402]: Good callout.

Comment (blakef) [reply packages/cli/src/commands/init/init.ts]: I've done a bunch more work on this as well as getting some feedback  in #2345. 
- The template version and react-version can be set independently on `init`.
- By default they'll both select `latest`.
- If you know best, you can avoid any of the measures I've put in and override with `--template <whatever your template is>@<whatever version you want>`. 
- `--template-version` is now a shortcut for `--template @react-native-community/template@<version>`
- `--version` is a shortcut specifically for the version of react-native.

If the new template isn't available, the behaviour is exactly was it was before.

File: docs/commands.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ React Native CLI comes with following commands:
 
 > Available since 0.60.0
 
-> **IMPORTANT**: Please note that this command is not available through `react-native-cli`, hence you need to either invoke it directly from `@react-native-community/cli` or `react-native` package which proxies binary to this CLI since 0.60.0, so it's possible to use it with e.g. `npx`.
+> **IMPORTANT**: Please note that this command is not available through `react-native-cli`, hence you need to either invoke it directly from the `@react-native-community/cli`  package which proxies binary to this CLI since 0.60.0, so it's possible to use it with e.g. `npx`.
```
Comment (cortinico) [inline docs/commands.md:25]:
```suggestion
> **IMPORTANT**: Please note that this command is not available through `react-native-cli`, hence you need to invoke it directly from the `@react-native-community/cli` package if you want to use it with e.g. `npx`.
```

File: docs/commands.md [hunk 3]
```diff
@@ -46,6 +46,10 @@ Uses a custom directory instead of `<projectName>`.
 
 Uses a custom app title instead of `<projectName>`.
 
+#### `--template-version <string>
+
+Use a specific version of the [Community Template](https://github.com/react-native-community/template).  This is a shortcut for `--template @react-native-community/template@<version>`. It will default to the latest version if the `--template-version` argument isn't provided.
```
Comment (cortinico) [inline docs/commands.md:51]: Do we expect users to use this? It seems unnecessary at first, but as we have it right now it's ok to keep it

**Review by cortinico — APPROVED** [review]
LGTM 👍
@szymonrybczak could you also do a pass?

File: docs/commands.md (hunk 3 above)
Comment (blakef) [reply docs/commands.md:51]: It can be stripped because we can achieve the same effect with `--template`.

Comment (cortinico) [reply docs/commands.md]: If possible, I won't expose it at first. It's yet another flag/api to maintain. We can still expose it if a user needs it afterwards

**Review by szymonrybczak — APPROVED** [review]

**Pull request merged**